cpp/src/io/orc/orc.hpp

```diff
@@ -528,20 +531,28 @@ class ProtobufWriter {
                            TypeKind kind,
                            ColStatsBlob const* stats);
 
+  void resize(std::size_t bytes) { m_buf->resize(bytes); }
+
+  std::size_t size() const { return m_buf->size(); }
+  uint8_t const* data() { return m_buf->data(); }
```

Comment:
could these be combined into a span?

Reply:
Unfortunately no.
 * We need them to pass into `host_write`.
 * In addition, we also need to access `size()` separately for data processing.